lib/broccoli/ember-app.js

```diff
@@ -1176,8 +1191,19 @@ EmberApp.prototype.testFiles = function(coreTestTree) {
 
   var emberCLITree = this._processedEmberCLITree();
 
-  var testJs = this.concatFiles(mergeTrees([external, coreTestTree]), {
-    inputFiles: this.legacyTestFilesToAppend,
+  var coreExt = mergeTrees([external, coreTestTree])
+
+  coreExt = stewMap(coreExt, function(content, relativePath){
+    if (legacyTestFilesToAppend.indexOf(relativePath) > -1) {
```

Comment:
Ya, that would work.  Instead of prepending each file, we would just concat a `before-document-guard.js` file, the file contents, then an `after-document-guard.js` file.  That would properly preserve sourcemaps.